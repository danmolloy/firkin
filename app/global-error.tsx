"use client";

import * as Sentry from "@sentry/nextjs";
import Error from "next/error";

import { useEffect } from "react";

export default function GlobalError({ error }: { error: Error }) {
  useEffect(() => {
    Sentry.captureException(error);
  }, [error]);

  return (
    <html lang="en">
      <body>
       <p>Error!</p>
      </body>
    </html>
  );
}