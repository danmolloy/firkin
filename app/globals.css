@tailwind base;
@tailwind components;
@tailwind utilities;


.mix-text {
  mix-blend-mode: difference;
};