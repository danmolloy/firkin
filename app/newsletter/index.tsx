'use client'
import { Formik, Field, Form, ErrorMessage } from 'formik';
import * as Yup from 'yup';
import { useEffect, useState } from 'react';
import axios from "axios";

export default function NewsletterIndex() {
  const [sendStatus, setSendStatus] = useState<"success"|"sending"|"err"|null>(null)

  const sendFail = (<div><h2 className="text-2xl">Sign up failed.</h2> <p>Please try again or contact us.</p></div>)

  const sendSuccess = (<div><h2 className="text-2xl">Sign up successful!</h2></div>)

  const sendingMsg = (<div><h2 className="text-2xl">Submitting...</h2></div>)

  return (
      <div data-testid="newsletter-form" id="dark-section" className=' px-8 py-8 flex flex-col bg-white text-black'>
              <div className='border border-zinc-400 shadow rounded md:w-4/5 lg:w-1/2 p-2'>
              <h2 className="text-2xl">Join our Mailing List</h2>
              <div>
                <p className='text-gray-500'>Get info on upcoming gigs and events.</p>
              </div>

    <Formik    
    initialValues={{
      email: '',
    }}
    validationSchema={Yup.object({
      email: Yup.string()
        .email('Invalid email address')
        .required('email required'),
    })}
    onSubmit={async (values, actions ) => {
      setSendStatus("sending")
      await new Promise(resolve => setTimeout(resolve, 1000))
      axios.post("/newsletter/api", values)
      .then(async (res) => {
        console.log('Response received')
        if (res.status === 200) {
          setSendStatus("success")
          actions.setSubmitting(false)
          actions.resetForm()
        } else {
          actions.setSubmitting(false)
          setSendStatus("err")
        }
        }).catch(e => {
          actions.setSubmitting(false)
          setSendStatus("err")
        })
       }}> 
      {(props) => (
      <Form  className=" flex flex-col  ">

          
          <div className="flex flex-col m-2">
          <label htmlFor='email' className="form-label ">Email</label>
          <Field
            id="email" 
            className="text-black border border-zinc-400 rounded-md w-full p-1 " 
            type="email"
            name="email"/>
            <div className="h-6">
          <ErrorMessage name="email" >
          { msg => <div className="text-sm text-red-500 h-6">{msg}</div> }
          </ErrorMessage>
          </div>
          </div>
          
        <button disabled={props.isSubmitting || sendStatus === "sending"} id="submit-button" type='submit' className="disabled:opacity-30 hover:bg-blue-100 border shadow-sm border-blue-500 text-blue-500 m-2 p-1 w-24 rounded self-end">Submit</button>
        <div>
        {sendStatus === "sending" 
        ? sendingMsg
        : sendStatus === "err" 
        ? sendFail
        : sendStatus === "success" 
        ? sendSuccess
        : null}
      </div>
      </Form> )}
    </Formik>
    </div>
    </div>
  )
}