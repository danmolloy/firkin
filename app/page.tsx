import About from "@/components/about";
import Calendar from "@/components/calendar";
import ContactForm from "@/components/contactForm";
import EventTile from "@/components/eventTile";
import Footer from "@/components/footer";
import Header from "@/components/header";
import Hero from "@/components/hero";
import Musicians from "@/components/musicians";
import { DateTime } from "luxon";

async function getData() {
  const API_KEY = process.env.API_KEY
      const CALENDAR_ID = process.env.CALENDAR_ID
      const res = await fetch(
        `https://www.googleapis.com/calendar/v3/calendars/${CALENDAR_ID}/events?key=${API_KEY}`,
        {cache: "no-store"}
      );
 
  if (!res.ok) {
    // This will activate the closest `error.js` Error Boundary
    throw new Error('Failed to fetch data')
  }
 
  return res.json()
}

export type Event = {
  id: string
  summary: string
  location: string
  description: string
  start: {
    dateTime: Date
  }
  end: {
    dateTime: Date
  }
}

export default async function Home() {
  //const [showHeader, setShowHeader] = useState<boolean>(true)
  //const [showMenu, setShowMenu] = useState<boolean>(false)
  const data = await getData()
  const calendar = data.items.sort((a:Event, b:Event) => 
    new Date(a.start.dateTime).getTime() - new Date(b.start.dateTime).getTime()).filter((i: Event) => 
      DateTime.fromJSDate(new Date(i.start.dateTime)) >= DateTime.now())

  return (
    <main className="flex flex-col text-black bg-slate-50 " data-testid="main-page">
      <Header /* setShowMenu={() => setShowMenu(!showMenu)} setShowHeader={(arg) => setShowHeader(arg)} *//>
      <Hero nextGig={calendar[0]} />
      <div id="light" className="z-10 font-text ">
        <div id="upper" className="flex flex-col">
          <About />
          <Calendar events={calendar.length > 0 ? calendar : []} />
        </div>
        <Musicians />
        <ContactForm />
        <Footer />
      </div>
    </main>
  );
}
