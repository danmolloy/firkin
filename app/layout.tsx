import type { Metadata } from "next";
import { Unkempt, Quicksand } from "next/font/google";
import "./globals.css";
import { Analytics } from "@vercel/analytics/react"

const quicksand = Quicksand({weight: ["300", "400", "500", "600", "700"], subsets: ["latin"], variable: '--font-quicksand' })
const title = Unkempt({weight: "400", subsets: ["latin"], variable: '--font-unkempt'})
export const metadata: Metadata = {
  title: "The Big Firkin Band",
  description: "17-piece jazz big band sensation in South East London. Live at the Fox & Firkin on the second Sunday of every month 3:30pm-6pm!",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en" className={`${title.variable}  ${quicksand.variable}`}>
      <body >
        {children}
      </body>
      {process.env.ENVIRONMENT !== "local" && <Analytics />}
    </html>
  );
}
