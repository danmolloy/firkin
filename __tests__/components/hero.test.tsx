import Hero, { HeroProps } from "@/components/hero"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import { Event } from "@/app/page"
import { DateTime } from "luxon"

export const mockEvent: Event = {
    summary: "mockSummary",
    description: "mockDescription",
    id: "mockId",
    location: "FoxNFirkin",
    start: {
      dateTime: new Date("July 31, 2009 01:43:00")
    },
    end: {
      dateTime: new Date("July 31, 2009 01:44:00")
    }
  }


const mockProps: HeroProps = {
  nextGig: mockEvent
}

describe("<Hero />", () => {
  beforeEach(() => {
    render(<Hero {...mockProps} />)
  })
  it("hero-section is in the document", () => {
    const heroSection = screen.getByTestId("hero-section")
    expect(heroSection).toBeInTheDocument()
  })
  it("next gig details is in the document", () => {
    const gigTile = screen.getByTestId(`next-gig`)
    expect(gigTile).toBeInTheDocument()
    expect(gigTile.textContent).toMatch("Next gig:")
    expect(gigTile.textContent).toMatch(`${DateTime.fromJSDate(new Date(mockProps.nextGig.start.dateTime)).setZone('Europe/London').toFormat("h:mm a")} @ ${mockProps.nextGig.location.split(",")[0]}`)
    expect(gigTile).toHaveAttribute("href", "/#calendar")
  })
  it("band name is in the document", () => {
    const heroSection = screen.getByTestId("hero-section")
    expect(heroSection.textContent).toMatch("The Big Firkin Band")
  })
  it("tagline or quote is in the document", () => {
    const heroSection = screen.getByTestId("hero-section")
    expect(heroSection.textContent).toMatch("17-piece jazz big band sensation")
    expect(heroSection.textContent).toMatch("Live at the Fox & Firkin on the second Sunday of the month 3:30pm-6pm!")

  })
  it("instagram link is in the document", () => {
    const instaLink = screen.getByTestId("insta-link")
    expect(instaLink).toBeInTheDocument()
    expect(instaLink.textContent).toMatch("Follow us")
    expect(instaLink).toHaveAttribute("href", "https://www.instagram.com/thebigfirkinband/")
    expect(instaLink).toHaveAttribute("target", "_blank")
  })
  it("matches snapshot", () => {
    const heroSection = screen.getByTestId("hero-section")
    expect(heroSection).toMatchSnapshot()
  })
})