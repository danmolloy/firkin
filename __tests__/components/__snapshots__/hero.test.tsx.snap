// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> matches snapshot 1`] = `
<div
  class="bg-black text-white pt-12 text-lg pb-8 px-4 flex flex-col items-center justify-between h-screen fixed w-screen  z-0"
  data-testid="hero-section"
>
  <div
    class="text-center flex flex-col items-center mt-4 md:mt-24 self-center lg:w-1/2"
  >
    <h1
      class="font-display text-6xl md:text-7xl my-2 md:mt-12 text-yellow-500"
    >
      The Big Firkin Band
    </h1>
    <div
      class=" p-2 md:p-4"
    >
      <p
        class="text-xl font-text border-b pb-4 text-gray-300"
      >
        17-piece jazz big band sensation
      </p>
      <p
        class="text-base md:text-xl font-text  pt-2 text-yellow-400"
      >
        Live at the Fox & Firkin on the second Sunday of the month 3:30pm-6pm!
      </p>
    </div>
  </div>
  <div
    class="self-start  md:text-2xl mb-20 px-2 pt-1 md:pt-2 font-text"
  >
    <a
      class="hover:cursor-pointer hover:text-gray-200"
      data-testid="next-gig"
      href="/#calendar"
    >
      <h3
        class=""
      >
        Next gig:
      </h3>
      <p
        class=" md:text-3xl text-xl"
        data-testid="start-time"
      >
        31 Jul 2009
      </p>
      <div
        class="flex flex-row items-center"
      >
        <p
          data-testid="start-time"
        >
          01:43 AM
           @ 
          FoxNFirkin
        </p>
      </div>
    </a>
    <div
      class="flex flex-row mb-2 mt-2 md:mt-4"
    >
      <a
        class="text-gray-400 hover:text-yellow-500 flex flex-row items-center "
        data-testid="insta-link"
        href="https://www.instagram.com/thebigfirkinband/"
        target="_blank"
        title="Firkin Band Instagram"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
          />
        </svg>
        <p
          class="text-sm mx-2"
        >
          Follow us
        </p>
      </a>
    </div>
  </div>
</div>
`;
